api/request.go

```diff
@@ -41,16 +42,23 @@ type Normalizable interface {
 }
 
 func (r *RequestedResource) NormalizeConnection() error {
-	blob := strings.Split(strings.TrimSpace(r.Vds), "?")
-	r.Vds = blob[0]
+	url, err := url.Parse(r.Vds)
+	if  err  != nil {
+		return core.NewInvalidArgument(err.Error())
+	}
+	if url.Hostname() != "" && url.Scheme != "" {
```

Comment:
Maybe we can do the same logic a bit more straighforward? Allow the library to gather the string back

```
	if strings.TrimSpace(r.Sas) == "" {
		if url.RawQuery == "" {
		    return core.NewInvalidArgument("No valid Sas token is found in the request")
		}
		r.Sas = url.RawQuery
	}

	url.RawQuery = ""
	url.Host = url.Hostname()
	r.Vds = url.String()
```

Now it is not clear why we need to check hostname and scheme for emptiness.

Reply:
I will replace the code with the logic you suggested that is better 😊